### PR TITLE
Add `fmin`/`fmax` functions to `math.h`

### LINKED ISSUE #299: Implement libm min and max functions
In RUST cast from float to int (with `as` keyword) is saturating operation: https://doc.rust-lang.org/reference/expressions/operator-expr.html#numeric-cast

and for now it generates following error:

```
LLVM ERROR: unable to legalize instruction: %22:_(s32) = G_INTRINSIC intrinsic(@llvm.fptosi.sat), %8:_(s32) (in function: main)
```

### PR DESCRIPTION
**Pull request by mlund**
This adds min and max functions for `float` and `double`.
Closes #299.

### CODE REVIEW

File: mos-platform/common/c/math.cc [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#include <math.h>
+
+template <typename T> static inline bool _is_nan(const T x) { return x != x; }
```
Comment (mysterymath) [inline mos-platform/common/c/math.cc:3]: May as well actually implement the macro in `<math.h>`:

`#define isnan(arg) __builtin_isfpclass(arg, __FPCLASS_SNAN | __FPCLASS_QNAN)`

File: mos-platform/common/c/math.cc [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+template <typename T> static inline T _fmax(const T x, const T y) {
+  if (_is_nan(x)) return y;
```
Comment (mysterymath) [inline mos-platform/common/c/math.cc:6]: This isn't needed; if `x` is nan, then `x > y` is false, so the ternary returns `y`. Replace this line with a comment to this effect.

File: mos-platform/common/c/math.cc [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  if (_is_nan(y)) return x;
+  return x > y ? x : y;
+}
+
+template <typename T> static inline T _fmin(const T x, const T y) {
+  if (_is_nan(x)) return y;
```
Comment (mysterymath) [inline mos-platform/common/c/math.cc:12]: Same here.

File: mos-platform/common/c/math.cc (hunk 2 above)
Comment (mlund) [reply mos-platform/common/c/math.cc:6]: Nice catch - fixed.

File: mos-platform/common/c/math.cc (hunk 3 above)
Comment (mlund) [reply mos-platform/common/c/math.cc:12]: fixed

File: mos-platform/common/c/math.cc (hunk 1 above)
Comment (mlund) [reply mos-platform/common/c/math.cc:3]: Or use the existing `__builtin_isnan`?

Comment (mlund) [reply mos-platform/common/c/math.cc]: Also, I now notice that `__builtin_fmin` and `__builtin_fmax` along with several other math functions are already defined and seem to be correct. Perhaps better to use these? Are these covered by existing tests in the separate test suite?

Comment (mysterymath) [reply mos-platform/common/c/math.cc]: Yeah, `__builtin_isnan` lowers to the same thing and is clearer.

`__builtin_fmin` doesn't help us; it lowers to a compiler intrinsic that most naturally lowers to a call to `fmin`. By comparison, `__builtin_isnan` lowers to a call to `__unordsf2` in `compiler_rt`, plus a bit of additional logic.

**Review by mysterymath — APPROVED** [review]
LGTM!

**Pull request merged**